ui/js/controllers/perf/compare.js

```diff
@@ -853,3 +866,102 @@ perf.controller('CompareSubtestResultsCtrl', [
             });
         });
     }]);
+
+perf.controller('CompareSubtestDistributionCtrl', ['$scope', '$stateParams', '$q', 'ThRepositoryModel',
+    'PhSeries', 'ThResultSetModel', 'metricsgraphics',
+    function CompareSubtestDistributionCtrl($scope, $stateParams, $q, ThRepositoryModel,
+        PhSeries, ThResultSetModel, metricsgraphics) {
+        $scope.originalRevision = $stateParams.originalRevision;
+        $scope.newRevision = $stateParams.newRevision;
+        $scope.originalSubtestSignature = $stateParams.originalSubtestSignature;
+        $scope.newSubtestSignature = $stateParams.newSubtestSignature;
+        $scope.dataLoading = true;
+        let loadRepositories = ThRepositoryModel.load();
+        const drawReplicateGraph = function (project, revision, subtestSignature, target) {
+            //let resultSet;
+            let replicateData = {};
+            return ThResultSetModel.getResultSetsFromRevision(project, revision).then(
+                (revisionData) => {
+                    replicateData.resultSet = revisionData[0];
+                    return PhSeries.getSeriesData(project, {
+                        signatures: subtestSignature,
+                        push_id: replicateData.resultSet.id
+                    });
+                }).then((perfDatumList) => {
+                    if (!perfDatumList[subtestSignature]) {
+                        replicateData.replicateDataError = true;
+                        return;
+                    }
+                    let numRuns = perfDatumList[subtestSignature].length;
+                    let replicatePromises = perfDatumList[subtestSignature].map(
+                        (value) => {return PhSeries.getReplicateData({job_id: value.job_id});
+                        });
+                    return $q.all(replicatePromises).then((replicateData) => {
+                        let replicateValues = replicateData.concat.apply([],
+                                replicateData.map(function (data) {
+                                    let testSuite = data.suites.find((suite) => {
+                                        return suite.name === $scope.testSuite;
+                                    });
+                                    let subtest = testSuite.subtests.find((subtest) =>{
+                                        return subtest.name === $scope.subtest;
+                                    });
+                                    return subtest.replicates;
+                                })
+                            );
+                        replicateValues = replicateValues.map((value, index) => {
+                            return {"replicate": (index + 1).toString(), "value": value};
```

Comment:
If I send data with replicate value zero, i.e.,

[{"replicate" : "0", "value": 3.5}, ...], metrics-graphics throwas the following error

TypeError: args.scales.YGROUP is not a function
at mg_bar_add_zero_line (metricsgraphics.js:2718)
at Object.mgDrawAxis.numerical (metricsgraphics.js:2571)
at axisFactory.draw (metricsgraphics.js:2613)
at barChart.init (metricsgraphics.js:6362)
at new barChart (metricsgraphics.js:6899)
at Object.MG.data_graphic (metricsgraphics.js:836)
at $q.all.then.replicateData (compare.js:751)
at processQueue (angular.js:16696)
at eval (angular.js:16712)
at Scope.$eval (angular.js:17994)